contracts/stabilizer/EthPegStabilityModule.sol

```diff
@@ -0,0 +1,90 @@
+pragma solidity ^0.8.4;
+
+import "./PegStabilityModule.sol";
+
+contract EthPegStabilityModule is PegStabilityModule {
+    using SafeERC20 for IERC20;
+
+    constructor(
+        address _coreAddress,
+        address _oracleAddress,
+        uint256 _mintFeeBasisPoints,
+        uint256 _redeemFeeBasisPoints,
+        uint256 _reservesThreshold,
+        uint256 _feiLimitPerSecond,
+        uint256 _mintingBufferCap,
+        int256 _decimalsNormalizer,
+        bool _doInvert,
+        IFei _FEI
+    ) PegStabilityModule(
+        _coreAddress,
+        _oracleAddress,
+        _mintFeeBasisPoints,
+        _redeemFeeBasisPoints,
+        _reservesThreshold,
+        _feiLimitPerSecond,
+        _mintingBufferCap,
+        _decimalsNormalizer,
+        _doInvert,
+        IERC20(address(0)), /// since the token for this PSM is eth, the address is 0
+        _FEI
+    ) {}
+
+    /// @notice function to redeem FEI for an underlying asset
+    function redeem(address to, uint256 amountFeiIn) external override nonReentrant whenNotPaused returns (uint256 amountEthOut) {
+        updateOracle();
+
+        amountEthOut = getRedeemAmountOut(amountFeiIn);
+        FEI.transferFrom(msg.sender, address(this), amountFeiIn);
+        FEI.burn(amountFeiIn);
+
+        Address.sendValue(payable(to), amountEthOut);
+
+        emit Redeem(to, amountFeiIn);
+    }
+
+    /// @notice function to mint Fei by sending eth
+    function mint(address to, uint256 amountIn) external payable override nonReentrant whenNotPaused returns (uint256 amountFeiOut) {
+        require(msg.value == amountIn, "EthPegStabilityModule: Sent value does not equal input");
+
+        updateOracle();
+
+        amountFeiOut = getMintAmountOut(amountIn);
+
+        _mintFei(msg.sender, amountFeiOut);
+
+        emit Mint(to, amountIn);        
+    }
```

Comment:
These should leverage internal methods and virtual methods to share more logic with the ERC20PSM and PSM. See the way bonding curve and reserve stabilizer handled this inheritance pattern.